core/src/test/java/io/questdb/test/griffin/engine/functions/catalogue/FunctionListFunctionFactoryTest.java

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin.engine.functions.catalogue;
+
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.FunctionFactoryDescriptor;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.engine.functions.catalogue.FunctionListFunctionFactory;
+import io.questdb.test.AbstractGriffinTest;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static io.questdb.griffin.engine.functions.catalogue.FunctionListFunctionFactory.isExcluded;
+
+public class FunctionListFunctionFactoryTest extends AbstractGriffinTest {
+
+    @Test
+    public void testFunctions() throws Exception {
+        TestUtils.printSql(compiler, sqlExecutionContext, "functions()", sink);
+        Assert.assertEquals(expectedFunctions(), extractFunctionsFromSink());
+    }
+
+    @Test
+    public void testFunctionsWithFilter() throws Exception {
+        TestUtils.printSql(
+                compiler,
+                sqlExecutionContext,
+                "SELECT name FROM (SELECT name, count(name) FROM functions() GROUP BY name, type ORDER BY name)",
+                sink
+        );
+        Assert.assertEquals(expectedFunctionNames(), extractFunctionNamesFromSink());
+    }
+
+    private static Set<String> expectedFunctionNames() {
+        Set<String> names = new HashSet<>();
+        for (FunctionFactory factory : ServiceLoader.load(FunctionFactory.class, FunctionFactory.class.getClassLoader())) {
+            String signature = factory.getSignature();
+            String name = signature.substring(0, signature.indexOf('('));
+            if (isExcluded(name)) {
+                continue;
+            }
+            names.add(name);
+        }
+        return names;
+    }
```

Comment:
It'd be good to add a test with concrete expected output (maybe just for some functions ) .

Reply:
addressed in a1a73ca10077447f80b5c96f5ce3d8553ceb85f5 in `FunctionFactoryDescriptorTest`